app.rb

```diff
@@ -0,0 +1,162 @@
+require './book'
+require './rental'
+require './student'
+require './teacher'
+require './main'
+require './list_rentals'
+require './rent_out_book'
+
+class App
+  attr_accessor :books, :persons, :rentals
+
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def select_option
+    option = gets.chomp
+
+    case option
+    when '1' then list_all_books
+    when '2' then list_all_people
+    when '3' then create_a_person
+    when '4' then create_a_book
+    when '5' then create_a_rental
+    when '6' then list_rentals
+    when '7' then stop_application
+    else
+      puts 'Invalid option. Option must be between 1 and 7'
+      select_option
+    end
+  end
+  # rubocop:enable Metrics/CyclomaticComplexity
+
+  def library_menu
+    puts ''
+    puts 'Welcome to School library App!'
+    puts ''
+    puts 'Please chose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+    puts ''
+    select_option
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts 'No books available at the moment.'
+      puts ''
+      puts 'Enter 4 to create a new book'
+      select_option
+    else
+      puts 'List of available books:'
+      puts ''
+      @books.each do |book|
+        puts "Title: #{book.title}, Author: #{book.author}"
+      end
+      library_menu
+    end
+  end
+
+  def list_all_people
+    if @persons.empty?
+      puts 'No people available at the moment'
+      puts ''
+      puts 'Enter 3 to create a new person'
+      select_option
+    else
+      puts 'List of available people:'
+      puts ''
+      @persons.each do |person|
+        puts "#{[person.class]} Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+      library_menu
+    end
+  end
+
+  def create_a_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    @books.push(Book.new(title, author))
+    puts 'Book successfully created!'
+    library_menu
+  end
+
+  def create_a_person
+    puts 'Any person do you want to create ?'
+    puts '1 - Student'
+    puts '2 - Teacher'
+    print 'Input number : '
+
+    person_type = Integer(gets.chomp)
+
+    case person_type
+    when 1
+      create_a_student
```

Comment:
- [ ] You did a good job working on the suggested changes! 💯 However, there are still errors that need to be fixed in order to continue to the next milestone:

When I try to create a `student` or a `teacher` I get this error:

![image](https://github.com/aima98/OOP-School-Library/assets/77069011/9dffc646-2557-4c2d-a479-a43213aca882)
 
This is because when you create a person, you are pushing the instance to a variable called `people`. However, this variable doesn't exist. Maybe you meant `@persons` which is the one you initialized in your `App.rb` file. 🤓 🔧 


- [ ] Also, when this error is fixed, a new one appears:

![image](https://github.com/aima98/OOP-School-Library/assets/77069011/f794f85d-31d6-4169-b5bf-07620be208c2)

This happens because you are sending an argument incorrectly:

```rb
# app.rb line 136.
# Notice the "false" boolean at the end
@persons << Student.new(nil, age, name, false)

# student.rb
# Your initialize method expects a named parameter "parent_permission"
def initialize(classroom, age, name = 'Unknown', parent_permission: true)
  super(age, name, parent_permission: parent_permission)
  @classroom = classroom
end
```

To fix this error, simply pass the boolean as a named argument:

```rb
# app.rb line 136.
# Notice the change in the last argument
@persons << Student.new(nil, age, name, parent_permission: false)
```